ports/esp32/esp_error.c

```diff
@@ -0,0 +1,148 @@
+
+ESP_EXCEPTIONS(pcnt_event_disable(self->unit, evt_type))
+
+// is equivalent to
+
+esp_err_t err = pcnt_event_disable(self->unit, evt_type);
+if (err != ESP_OK)
+    mp_raise_EspError(err);
+*/
+
+// Defining classes
+// class EspError(Exception):
+typedef struct _mp_obj_esp_err_EspError_t {
+    mp_obj_base_t base;
+} mp_obj_esp_err_EspError_t;
+
+const mp_obj_type_t mp_type_EspError;
+/*
+// Defining EspError methods
+// def EspError.__init__(self, error_code: int=0, error_msg: str='')
+STATIC mp_obj_t esp_err_EspError_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *args) {
+    mp_arg_check_num(n_args, n_kw, 0, 2, false);
+    self->base.type = &mp_type_EspError;
+
+	mp_obj_esp_err_EspError_t *self = MP_OBJ_TO_PTR(args[0]);
+	mp_int_t error_code = mp_obj_get_int(args[1]);
+	const char* error_msg = mp_obj_str_get_str(args[2]);
+
+	//TODO: Your code here
+    return MP_OBJ_FROM_PTR(self);
+}
+*/
+/*
+STATIC mp_obj_t esp_err_EspError_print(const mp_print_t *print, mp_obj_t self_obj, mp_print_kind_t kind) {
+    esp_err_EspError_obj_t *self = MP_OBJ_TO_PTR(self_obj);
+    mp_printf(print, "EspError()");
+
+	//TODO: Your code here
+}
+*/
+
+STATIC mp_obj_t esp_err_EspError_esp_err_to_name(mp_obj_t self_obj, mp_obj_t err_obj) {
+	//mp_obj_esp_err_EspError_t *self = MP_OBJ_TO_PTR(self_obj);
+	mp_int_t err = mp_obj_get_int(err_obj);
+
+    const char *s = esp_err_to_name(err);
+
+    return mp_obj_new_str(s, strlen(s));
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_2(esp_err_EspError_esp_err_to_name_obj, esp_err_EspError_esp_err_to_name);
+
+// EspError stuff
+// Register class methods
+STATIC const mp_rom_map_elem_t esp_err_EspError_locals_dict_table[] = {
+    //{ MP_ROM_QSTR(MP_QSTR_ESP_ERR_FLASH_BASE), MP_ROM_INT(ESP_ERR_FLASH_BASE) },
+    { MP_ROM_QSTR(MP_QSTR_ESP_ERR_INVALID_ARG), MP_ROM_INT(ESP_ERR_INVALID_ARG) },
+    { MP_ROM_QSTR(MP_QSTR_ESP_ERR_INVALID_CRC), MP_ROM_INT(ESP_ERR_INVALID_CRC) },
+    { MP_ROM_QSTR(MP_QSTR_ESP_ERR_INVALID_MAC), MP_ROM_INT(ESP_ERR_INVALID_MAC) },
+    { MP_ROM_QSTR(MP_QSTR_ESP_ERR_INVALID_RESPONSE), MP_ROM_INT(ESP_ERR_INVALID_RESPONSE) },
+    { MP_ROM_QSTR(MP_QSTR_ESP_ERR_INVALID_SIZE), MP_ROM_INT(ESP_ERR_INVALID_SIZE) },
+    { MP_ROM_QSTR(MP_QSTR_ESP_ERR_INVALID_STATE), MP_ROM_INT(ESP_ERR_INVALID_STATE) },
+    { MP_ROM_QSTR(MP_QSTR_ESP_ERR_INVALID_VERSION), MP_ROM_INT(ESP_ERR_INVALID_VERSION) },
+    { MP_ROM_QSTR(MP_QSTR_ESP_ERR_MESH_BASE), MP_ROM_INT(ESP_ERR_MESH_BASE) },
+    { MP_ROM_QSTR(MP_QSTR_ESP_ERR_NOT_FOUND), MP_ROM_INT(ESP_ERR_NOT_FOUND) },
+    { MP_ROM_QSTR(MP_QSTR_ESP_ERR_NOT_SUPPORTED), MP_ROM_INT(ESP_ERR_NOT_SUPPORTED) },
+    { MP_ROM_QSTR(MP_QSTR_ESP_ERR_NO_MEM), MP_ROM_INT(ESP_ERR_NO_MEM) },
+    { MP_ROM_QSTR(MP_QSTR_ESP_ERR_TIMEOUT), MP_ROM_INT(ESP_ERR_TIMEOUT) },
+    { MP_ROM_QSTR(MP_QSTR_ESP_ERR_WIFI_BASE), MP_ROM_INT(ESP_ERR_WIFI_BASE) },
+    { MP_ROM_QSTR(MP_QSTR_ESP_FAIL), MP_ROM_INT(ESP_FAIL) },
+    { MP_ROM_QSTR(MP_QSTR_ESP_OK), MP_ROM_INT(ESP_OK) },
```

Comment:
I'm not sure if all(or part) ESP-IDF codes should be pulled(Q-stringed) as constants into MicroPython.
The complete list is very long and not in the "micro" paradigm.